app/domain/authentication/update_authenticator_config.rb

```diff
@@ -0,0 +1,60 @@
+require 'command_class'
+
+module Authentication
+
+  UpdateAuthenticatorConfig = CommandClass.new(
+    dependencies: {
+      webservice_class: ::Authentication::Webservice,
+      resource_class: ::Resource,
+      authenticator_config_class: ::AuthenticatorConfig
+    },
+    inputs: %i(account authenticator service_id enabled current_user)
+  ) do
+    
+    def call
+      validate_resource_visible
```

Comment:
a. can you elaborate on why do we need `validate_resource_visible`?
b. You can see my comments below but it seems that you can user here [ValidateAccountExists](https://github.com/cyberark/conjur/blob/aac94ad69d2d13e07ac9a90e5e9288e01121139b/app/domain/authentication/validate_account_exists.rb), [ValidateWebserviceExists](https://github.com/cyberark/conjur/blob/aac94ad69d2d13e07ac9a90e5e9288e01121139b/app/domain/authentication/validate_webservice_exists.rb) & [ValidateWebserviceAccess](https://github.com/cyberark/conjur/blob/aac94ad69d2d13e07ac9a90e5e9288e01121139b/app/domain/authentication/validate_webservice_access.rb)